worker.py

```
@@ -0,0 +1,193 @@
+            db, email_id, EmailStatus.FAILED,
+            error_message=error_msg,
+            increment_retry=True
+        )
+        
+        # Get updated email to check retry count
+        email = await email_service.get_by_id(db, email_id)
+        
+        if email.retry_count < settings.max_retries:
+            # Requeue for retry with exponential backoff
+            retry_delay = settings.retry_delay_seconds * (2 ** (email.retry_count - 1))
+            logger.info(f"Requeuing email {email_id} for retry in {retry_delay}s")
```

Ruff (0.14.10)

Logging statement uses f-string

(G004)